automatedtests/src/androidTest/java/com/mux/player/media3/automatedtests/BandwidthMetricTests.java

```diff
@@ -71,52 +79,93 @@ public void init() {
 
-//  @Test
-//  public void testBandwidthMetricsHls() {
-//    testBandwidthMetrics();
-//  }
-//
-//  @Test
-//  public void testBandwidthMetricsDash() {
-//    testBandwidthMetrics();
-//  }
-
-  public void testBandwidthMetrics() {
     try {
       if (!testActivity.waitForPlaybackToStart(waitForPlaybackToStartInMS)) {
         fail("Playback did not start in " + waitForPlaybackToStartInMS + " milliseconds !!!");
       }
+      long startTime = System.currentTimeMillis();
       for (int i = 0; i < manifestDelayList.length; i++) {
-        System.out.println("Waiting for segment number: " + i);
-        httpServer.setHLSManifestDelay(manifestDelayList[i]);
+        Log.i(TAG,"Waiting for segment number: " + i);
+        //httpServer.setHLSManifestDelay(manifestDelayList[i]);
+        httpServer.setAdditionalHeader("x-cdn", "cdn" + (i / 5));
         if (!httpServer.waitForNextSegmentToLoad(waitForPlaybackToStartInMS * 3)) {
           fail("HLS playback segment did not start in " + waitForPlaybackToStartInMS + " ms !!!");
         }
+        if (System.currentTimeMillis() - startTime > PLAY_PERIOD_IN_MS * 2) {
+          break;
+        }
       }
       testActivity.runOnUiThread(new Runnable() {
         public void run() {
           pView.getPlayer().stop();
         }
       });
-      checkMimeType();
-      ArrayList<JSONObject> requestCompletedEvents = networkRequest
-          .getAllEventsOfType(RequestCompleted.TYPE);
-      ArrayList<JSONObject> requestCanceledEvents = networkRequest
-          .getAllEventsOfType(RequestCanceled.TYPE);
-      ArrayList<JSONObject> requestFailedEvents = networkRequest
-          .getAllEventsOfType(RequestFailed.TYPE);
-      checkRequests(requestCompletedEvents,
-          true, false, false);
-      checkRequests(requestCanceledEvents,
-          false, true, false);
+
+      List<JSONObject> cdnChanges = networkRequest.getAllEventsOfType("cdnchange");
+      Log.i(TAG,"got events: " + networkRequest.getReceivedEventNames());
+      Log.i(TAG,"CDN changes: " + cdnChanges);
+
+      List<String> videoCdnValues = cdnChanges.stream()
+          .map(it -> it.optString(VideoData.VIDEO_CDN))
+          .collect(Collectors.toList());
+      List<String> videoPrevCdnValues = cdnChanges.stream()
+          .map(it -> it.optString(VideoData.VIDEO_PREVIOUS_CDN))
+          .collect(Collectors.toList());
+      assertEquals(
+          "First cdnchange should be for cdn0",
+          "cdn0", videoCdnValues.get(0)
+      );
+      assertTrue(
+          "First cdnchange should have no previous cdn value",
+          TextUtils.isEmpty(videoPrevCdnValues.get(0))
+      );
+      assertEquals(
+          "Second cdnchange should be for cdn1",
+          "cdn1", videoCdnValues.get(1)
+      );
+      assertEquals(
+          "Second cdnchange should have no previous cdn value",
+          "cdn0", videoPrevCdnValues.get(1)
+      );
```

Comment:
```suggestion
      assertEquals(
          "Second cdnchange should have the previous cdn value",
          "cdn0", videoPrevCdnValues.get(1)
      );
```
This tests the right thing but the description was off, right?